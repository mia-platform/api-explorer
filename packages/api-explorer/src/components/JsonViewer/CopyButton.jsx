import React from 'react'
import PropTypes from 'prop-types'
import {Button} from 'antd'
import {FormattedMessage} from 'react-intl'

import CopyText from '../CopyText'

function stringify (schema) {
    try {
      const stringified = JSON.stringify(schema)
      return stringified
    }catch(err) {
      return null
    }
}

export default function CopyButton ({schema}) {
    const stringifiedSchema = stringify(schema)

    if (!stringifiedSchema) {
        return null
    }

    return (
      <CopyText text={stringifiedSchema}>
        {({isCopied}) => (
          <Button>
            {isCopied ?
              <FormattedMessage id="code.copied" defaultMessage="Copied" /> :
              <FormattedMessage id="code.copy" defaultMessage="Copy" />
            }
          </Button>
        )}
      </CopyText>
    )
}

CopyButton.propTypes = {
  schema: PropTypes.object.isRequired,
};