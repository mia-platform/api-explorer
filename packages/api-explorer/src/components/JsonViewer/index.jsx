/* eslint-disable react/require-default-props */
import React, {useState} from 'react'
import { FormattedMessage } from 'react-intl'
import {Button} from 'antd'
import PropTypes from 'prop-types'
import ReactJson from 'react-json-view'

import colors from '../../colors'

const collapseButtonStyle = {
  position: 'absolute',
  top: 10,
  right: 10,
  zIndex: 99999
}

export default function JsonViewer({ schema, missingMessage }) {
  const [isCollapsed, setIsCollapsed] = useState(true)

  return (
    schema ? (
      <div style={{position: 'relative'}}>
        <Button style={collapseButtonStyle} onClick={() => setIsCollapsed(!isCollapsed)}>
          <FormattedMessage id={`schemas.${isCollapsed ? 'expand' : 'collapse'}`} />
        </Button>
        <ReactJson
          src={schema}
          collapsed={isCollapsed ? 1 : false}
          collapseStringsAfterLength={100}
          enableClipboard={false}
          name={null}
          displayDataTypes={false}
          displayObjectSize={false}
          style={{
            padding: '20px 10px',
            backgroundColor: colors.jsonViewerBackground,
            fontSize: '12px',
            overflow: 'visible',
            whiteSpace: 'pre-wrap',
            wordBreak: 'break-all'
          }}
        />
      </div>
    ) : <FormattedMessage id={missingMessage} />
  )
}

JsonViewer.propTypes = {
  schema: PropTypes.object,
  missingMessage: PropTypes.string.isRequired
}
