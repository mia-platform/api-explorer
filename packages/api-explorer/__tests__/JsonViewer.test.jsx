import React from 'react'
import { FormattedMessage } from 'react-intl';
import ReactTestUtils from 'react-dom/test-utils'
import { mountWithIntl, loadTranslationObject } from 'enzyme-react-intl'
import ReactJson from 'react-json-view'
import { Button } from 'antd';

import strings from '../i18n/en.json'
import JsonViewer from '../src/components/JsonViewer';

describe('JsonViewer', () => {
  loadTranslationObject({
    ...strings,
    'foo.id': 'missing schema'
  })

  test('if schema is defined renders ReactJson', () => {
    const props = {
      schema: {
        foo: 'bar'
      },
      missingMessage: 'foo.id'
    }
    const element = mountWithIntl(<JsonViewer {...props} />)
    const reactJson = element.find(ReactJson)
    expect(reactJson).toHaveLength(1)
    expect(reactJson.prop('src')).toEqual(props.schema)
    expect(element.getDOMNode()).toMatchSnapshot()
  })

  test('if schema is undefined renders missing message', () => {
    const props = {
      missingMessage: 'foo.id'
    }
    const element = mountWithIntl(<JsonViewer {...props} />)
    const formattedMessage = element.find(FormattedMessage)
    expect(formattedMessage).toHaveLength(1)
    expect(formattedMessage.prop('id')).toEqual(props.missingMessage)
  })

  describe('Expand - Collapse button', () => {
    const schema = {
      foo: {
        bar: {
          lorem: 'ipsum'
        },
        something: {
          else: {
            else: {
              else: 'example'
            }
          }
        }
      }
    }
    let element
    beforeEach(() => {
      element = mountWithIntl(<JsonViewer schema={schema} missingMessage={'missing'} />)
    })

    test('is collapsed as default', () => {
      checkCollapse(element, {isCollapsed: true})

      expect(element.getDOMNode()).toMatchSnapshot()
    })

    test('expand when click button', () => {
      clickExpandButton(element)
      checkCollapse(element, {isCollapsed: false})

      expect(element.getDOMNode()).toMatchSnapshot()
    })

    test('flow', () => {
      checkCollapse(element, {isCollapsed: true})

      clickExpandButton(element)
      checkCollapse(element, {isCollapsed: false})
      
      clickExpandButton(element)
      checkCollapse(element, {isCollapsed: true})
      
      clickExpandButton(element)
      checkCollapse(element, {isCollapsed: false})
    })
  })
})

function clickExpandButton(element) {
  ReactTestUtils.act(() => {
    element.find(Button).prop('onClick')()
  })
  element.update()
}

function checkCollapse(element, {isCollapsed}) {
  if (isCollapsed) {
    expect(element.find(ReactJson).prop('collapsed')).toEqual(1)
    expect(element.find(Button).find(FormattedMessage).prop('id')).toEqual('schemas.expand')
    return
  }
  expect(element.find(ReactJson).prop('collapsed')).toEqual(10)
  expect(element.find(Button).find(FormattedMessage).prop('id')).toEqual('schemas.collapse')
}