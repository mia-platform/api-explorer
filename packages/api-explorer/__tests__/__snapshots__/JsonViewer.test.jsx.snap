// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsonViewer Expand - Collapse button expand when click button 1`] = `
<div
  style="position: relative;"
>
  <div
    class="react-json-view"
    style="font-family: monospace; cursor: default; background-color: rgb(237, 237, 237); position: relative; padding: 20px 10px; font-size: 12px; overflow: visible; white-space: pre-wrap; word-break: break-all;"
  >
    <div
      class="pretty-json-container object-container"
    >
      <div
        class="object-content"
      >
        <div
          class="object-key-val"
        >
          <span>
            <span
              style="display: inline-block; cursor: pointer;"
            >
              <div
                class="icon-container"
                style="display: inline-block; width: 17px;"
              >
                <span
                  class="expanded-icon"
                >
                  <svg
                    fill="currentColor"
                    style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                    viewBox="0 0 15 15"
                  >
                    <path
                      d="M0 5l6 6 6-6z"
                    />
                  </svg>
                </span>
              </div>
              <span />
              <span
                style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
              >
                {
              </span>
            </span>
            <div
              class="object-meta-data"
              style="display: inline-block; padding: 0px 0px 0px 10px;"
            />
          </span>
          <div
            class="pushed-content object-container"
          >
            <div
              class="object-content"
              style="margin-left: 6px;"
            >
              <div
                class="object-key-val"
                style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
              >
                <span>
                  <span
                    style="display: inline-block; cursor: pointer;"
                  >
                    <div
                      class="icon-container"
                      style="display: inline-block; width: 17px;"
                    >
                      <span
                        class="expanded-icon"
                      >
                        <svg
                          fill="currentColor"
                          style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                          viewBox="0 0 15 15"
                        >
                          <path
                            d="M0 5l6 6 6-6z"
                          />
                        </svg>
                      </span>
                    </div>
                    <span
                      style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                    >
                      <span
                        class="object-key"
                      >
                        <span
                          style="vertical-align: top;"
                        >
                          "
                        </span>
                        <span>
                          foo
                        </span>
                        <span
                          style="vertical-align: top;"
                        >
                          "
                        </span>
                      </span>
                      <span
                        style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                      >
                        :
                      </span>
                    </span>
                    <span
                      style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                    >
                      {
                    </span>
                  </span>
                  <div
                    class="object-meta-data"
                    style="display: inline-block; padding: 0px 0px 0px 10px;"
                  />
                </span>
                <div
                  class="pushed-content object-container"
                  style=""
                >
                  <div
                    class="object-content"
                    style="margin-left: 6px;"
                  >
                    <div
                      class="object-key-val"
                      style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
                    >
                      <span>
                        <span
                          style="display: inline-block; cursor: pointer;"
                        >
                          <div
                            class="icon-container"
                            style="display: inline-block; width: 17px;"
                          >
                            <span
                              class="expanded-icon"
                            >
                              <svg
                                fill="currentColor"
                                style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                                viewBox="0 0 15 15"
                              >
                                <path
                                  d="M0 5l6 6 6-6z"
                                />
                              </svg>
                            </span>
                          </div>
                          <span
                            style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                          >
                            <span
                              class="object-key"
                            >
                              <span
                                style="vertical-align: top;"
                              >
                                "
                              </span>
                              <span>
                                bar
                              </span>
                              <span
                                style="vertical-align: top;"
                              >
                                "
                              </span>
                            </span>
                            <span
                              style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                            >
                              :
                            </span>
                          </span>
                          <span
                            style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                          >
                            {
                          </span>
                        </span>
                        <div
                          class="object-meta-data"
                          style="display: inline-block; padding: 0px 0px 0px 10px;"
                        />
                      </span>
                      <div
                        class="pushed-content object-container"
                      >
                        <div
                          class="object-content"
                          style="margin-left: 6px;"
                        >
                          <div
                            class="variable-row"
                            style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
                          >
                            <span>
                              <span
                                class="object-key"
                                style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                              >
                                <span
                                  style="vertical-align: top;"
                                >
                                  "
                                </span>
                                <span
                                  style="display: inline-block;"
                                >
                                  lorem
                                </span>
                                <span
                                  style="vertical-align: top;"
                                >
                                  "
                                </span>
                              </span>
                              <span
                                style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                              >
                                :
                              </span>
                            </span>
                            <div
                              class="variable-value"
                              style="display: inline-block; padding-right: 6px; position: relative; cursor: default;"
                            >
                              <div
                                style="display: inline-block; color: rgb(203, 75, 22);"
                              >
                                <span
                                  class="string-value"
                                  style="cursor: default;"
                                >
                                  "
                                  ipsum
                                  "
                                </span>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <span
                        class="brace-row"
                      >
                        <span
                          style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
                        >
                          }
                        </span>
                      </span>
                    </div>
                    <div
                      class="object-key-val"
                      style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
                    >
                      <span>
                        <span
                          style="display: inline-block; cursor: pointer;"
                        >
                          <div
                            class="icon-container"
                            style="display: inline-block; width: 17px;"
                          >
                            <span
                              class="expanded-icon"
                            >
                              <svg
                                fill="currentColor"
                                style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                                viewBox="0 0 15 15"
                              >
                                <path
                                  d="M0 5l6 6 6-6z"
                                />
                              </svg>
                            </span>
                          </div>
                          <span
                            style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                          >
                            <span
                              class="object-key"
                            >
                              <span
                                style="vertical-align: top;"
                              >
                                "
                              </span>
                              <span>
                                something
                              </span>
                              <span
                                style="vertical-align: top;"
                              >
                                "
                              </span>
                            </span>
                            <span
                              style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                            >
                              :
                            </span>
                          </span>
                          <span
                            style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                          >
                            {
                          </span>
                        </span>
                        <div
                          class="object-meta-data"
                          style="display: inline-block; padding: 0px 0px 0px 10px;"
                        />
                      </span>
                      <div
                        class="pushed-content object-container"
                      >
                        <div
                          class="object-content"
                          style="margin-left: 6px;"
                        >
                          <div
                            class="object-key-val"
                            style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
                          >
                            <span>
                              <span
                                style="display: inline-block; cursor: pointer;"
                              >
                                <div
                                  class="icon-container"
                                  style="display: inline-block; width: 17px;"
                                >
                                  <span
                                    class="expanded-icon"
                                  >
                                    <svg
                                      fill="currentColor"
                                      style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                                      viewBox="0 0 15 15"
                                    >
                                      <path
                                        d="M0 5l6 6 6-6z"
                                      />
                                    </svg>
                                  </span>
                                </div>
                                <span
                                  style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                                >
                                  <span
                                    class="object-key"
                                  >
                                    <span
                                      style="vertical-align: top;"
                                    >
                                      "
                                    </span>
                                    <span>
                                      else
                                    </span>
                                    <span
                                      style="vertical-align: top;"
                                    >
                                      "
                                    </span>
                                  </span>
                                  <span
                                    style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                                  >
                                    :
                                  </span>
                                </span>
                                <span
                                  style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                                >
                                  {
                                </span>
                              </span>
                              <div
                                class="object-meta-data"
                                style="display: inline-block; padding: 0px 0px 0px 10px;"
                              />
                            </span>
                            <div
                              class="pushed-content object-container"
                            >
                              <div
                                class="object-content"
                                style="margin-left: 6px;"
                              >
                                <div
                                  class="object-key-val"
                                  style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
                                >
                                  <span>
                                    <span
                                      style="display: inline-block; cursor: pointer;"
                                    >
                                      <div
                                        class="icon-container"
                                        style="display: inline-block; width: 17px;"
                                      >
                                        <span
                                          class="expanded-icon"
                                        >
                                          <svg
                                            fill="currentColor"
                                            style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                                            viewBox="0 0 15 15"
                                          >
                                            <path
                                              d="M0 5l6 6 6-6z"
                                            />
                                          </svg>
                                        </span>
                                      </div>
                                      <span
                                        style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                                      >
                                        <span
                                          class="object-key"
                                        >
                                          <span
                                            style="vertical-align: top;"
                                          >
                                            "
                                          </span>
                                          <span>
                                            else
                                          </span>
                                          <span
                                            style="vertical-align: top;"
                                          >
                                            "
                                          </span>
                                        </span>
                                        <span
                                          style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                                        >
                                          :
                                        </span>
                                      </span>
                                      <span
                                        style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                                      >
                                        {
                                      </span>
                                    </span>
                                    <div
                                      class="object-meta-data"
                                      style="display: inline-block; padding: 0px 0px 0px 10px;"
                                    />
                                  </span>
                                  <div
                                    class="pushed-content object-container"
                                  >
                                    <div
                                      class="object-content"
                                      style="margin-left: 6px;"
                                    >
                                      <div
                                        class="variable-row"
                                        style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
                                      >
                                        <span>
                                          <span
                                            class="object-key"
                                            style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                                          >
                                            <span
                                              style="vertical-align: top;"
                                            >
                                              "
                                            </span>
                                            <span
                                              style="display: inline-block;"
                                            >
                                              else
                                            </span>
                                            <span
                                              style="vertical-align: top;"
                                            >
                                              "
                                            </span>
                                          </span>
                                          <span
                                            style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                                          >
                                            :
                                          </span>
                                        </span>
                                        <div
                                          class="variable-value"
                                          style="display: inline-block; padding-right: 6px; position: relative; cursor: default;"
                                        >
                                          <div
                                            style="display: inline-block; color: rgb(203, 75, 22);"
                                          >
                                            <span
                                              class="string-value"
                                              style="cursor: default;"
                                            >
                                              "
                                              example
                                              "
                                            </span>
                                          </div>
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                  <span
                                    class="brace-row"
                                  >
                                    <span
                                      style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
                                    >
                                      }
                                    </span>
                                  </span>
                                </div>
                              </div>
                            </div>
                            <span
                              class="brace-row"
                            >
                              <span
                                style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
                              >
                                }
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                      <span
                        class="brace-row"
                      >
                        <span
                          style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
                        >
                          }
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
                <span
                  class="brace-row"
                >
                  <span
                    style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
                  >
                    }
                  </span>
                </span>
              </div>
            </div>
          </div>
          <span
            class="brace-row"
          >
            <span
              style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
            >
              }
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
  <button
    class="ant-btn"
    style="position: absolute; top: 10px; right: 10px;"
    type="button"
  >
    <span>
      Collapse
    </span>
  </button>
</div>
`;

exports[`JsonViewer Expand - Collapse button is collapsed as default 1`] = `
<div
  style="position: relative;"
>
  <div
    class="react-json-view"
    style="font-family: monospace; cursor: default; background-color: rgb(237, 237, 237); position: relative; padding: 20px 10px; font-size: 12px; overflow: visible; white-space: pre-wrap; word-break: break-all;"
  >
    <div
      class="pretty-json-container object-container"
    >
      <div
        class="object-content"
      >
        <div
          class="object-key-val"
        >
          <span>
            <span
              style="display: inline-block; cursor: pointer;"
            >
              <div
                class="icon-container"
                style="display: inline-block; width: 17px;"
              >
                <span
                  class="expanded-icon"
                >
                  <svg
                    fill="currentColor"
                    style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                    viewBox="0 0 15 15"
                  >
                    <path
                      d="M0 5l6 6 6-6z"
                    />
                  </svg>
                </span>
              </div>
              <span />
              <span
                style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
              >
                {
              </span>
            </span>
            <div
              class="object-meta-data"
              style="display: inline-block; padding: 0px 0px 0px 10px;"
            />
          </span>
          <div
            class="pushed-content object-container"
          >
            <div
              class="object-content"
              style="margin-left: 6px;"
            >
              <div
                class="object-key-val"
                style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
              >
                <span>
                  <span
                    style="display: inline-block; cursor: pointer;"
                  >
                    <div
                      class="icon-container"
                      style="display: inline-block; width: 17px;"
                    >
                      <span
                        class="collapsed-icon"
                      >
                        <svg
                          fill="currentColor"
                          style="vertical-align: top; color: rgb(42, 161, 152); height: 1em; width: 1em; padding-left: 2px;"
                          viewBox="0 0 15 15"
                        >
                          <path
                            d="M0 14l6-6-6-6z"
                          />
                        </svg>
                      </span>
                    </div>
                    <span
                      style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                    >
                      <span
                        class="object-key"
                      >
                        <span
                          style="vertical-align: top;"
                        >
                          "
                        </span>
                        <span>
                          foo
                        </span>
                        <span
                          style="vertical-align: top;"
                        >
                          "
                        </span>
                      </span>
                      <span
                        style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                      >
                        :
                      </span>
                    </span>
                    <span
                      style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
                    >
                      {
                    </span>
                  </span>
                </span>
                <div
                  class="node-ellipsis"
                  style="display: inline-block; color: rgb(203, 75, 22); font-size: 18px; line-height: 10px; cursor: pointer;"
                >
                  ...
                </div>
                <span
                  class="brace-row"
                >
                  <span
                    style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 0px;"
                  >
                    }
                  </span>
                  <div
                    class="object-meta-data"
                    style="display: inline-block; padding: 0px 0px 0px 10px;"
                  />
                </span>
              </div>
            </div>
          </div>
          <span
            class="brace-row"
          >
            <span
              style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
            >
              }
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
  <button
    class="ant-btn"
    style="position: absolute; top: 10px; right: 10px;"
    type="button"
  >
    <span>
      Expand all
    </span>
  </button>
</div>
`;

exports[`JsonViewer if schema is defined renders ReactJson 1`] = `
<div
  style="position: relative;"
>
  <div
    class="react-json-view"
    style="font-family: monospace; cursor: default; background-color: rgb(237, 237, 237); position: relative; padding: 20px 10px; font-size: 12px; overflow: visible; white-space: pre-wrap; word-break: break-all;"
  >
    <div
      class="pretty-json-container object-container"
    >
      <div
        class="object-content"
      >
        <div
          class="object-key-val"
        >
          <span>
            <span
              style="display: inline-block; cursor: pointer;"
            >
              <div
                class="icon-container"
                style="display: inline-block; width: 17px;"
              >
                <span
                  class="expanded-icon"
                >
                  <svg
                    fill="currentColor"
                    style="vertical-align: top; color: rgb(88, 110, 117); height: 1em; width: 1em; padding-left: 2px;"
                    viewBox="0 0 15 15"
                  >
                    <path
                      d="M0 5l6 6 6-6z"
                    />
                  </svg>
                </span>
              </div>
              <span />
              <span
                style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54);"
              >
                {
              </span>
            </span>
            <div
              class="object-meta-data"
              style="display: inline-block; padding: 0px 0px 0px 10px;"
            />
          </span>
          <div
            class="pushed-content object-container"
          >
            <div
              class="object-content"
              style="margin-left: 6px;"
            >
              <div
                class="variable-row"
                style="border-left: 1px solid rgb(235, 235, 235); padding: 3px 5px 3px 20px;"
              >
                <span>
                  <span
                    class="object-key"
                    style="display: inline-block; color: rgb(0, 43, 54); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;"
                  >
                    <span
                      style="vertical-align: top;"
                    >
                      "
                    </span>
                    <span
                      style="display: inline-block;"
                    >
                      foo
                    </span>
                    <span
                      style="vertical-align: top;"
                    >
                      "
                    </span>
                  </span>
                  <span
                    style="display: inline-block; margin: 0px 5px; color: rgb(0, 43, 54); vertical-align: top;"
                  >
                    :
                  </span>
                </span>
                <div
                  class="variable-value"
                  style="display: inline-block; padding-right: 6px; position: relative; cursor: default;"
                >
                  <div
                    style="display: inline-block; color: rgb(203, 75, 22);"
                  >
                    <span
                      class="string-value"
                      style="cursor: default;"
                    >
                      "
                      bar
                      "
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <span
            class="brace-row"
          >
            <span
              style="display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 43, 54); padding-left: 3px;"
            >
              }
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
  <button
    class="ant-btn"
    style="position: absolute; top: 10px; right: 10px;"
    type="button"
  >
    <span>
      Expand all
    </span>
  </button>
</div>
`;
